### PR TITLE
fix: ui/ux

### PR DESCRIPTION
**Pull request by beyond009**
https://mask.atlassian.net/browse/FW-194
https://mask.atlassian.net/browse/FW-145
https://mask.atlassian.net/browse/FW-185
https://mask.atlassian.net/browse/FW-257
https://mask.atlassian.net/browse/FW-162
https://mask.atlassian.net/browse/FW-170
https://mask.atlassian.net/browse/FW-78
https://mask.atlassian.net/browse/FW-253
https://mask.atlassian.net/browse/FW-189
https://mask.atlassian.net/browse/FW-161
https://mask.atlassian.net/browse/FW-228
https://mask.atlassian.net/browse/FW-168
https://mask.atlassian.net/browse/FW-173

### CODE REVIEW

File: src/app/(normal)/profile/[source]/[id]/page.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import { safeUnreachable } from '@masknet/kit';
 import { useQuery } from '@tanstack/react-query';
 import { notFound } from 'next/navigation.js';
-import { useMemo } from 'react';
+import { Suspense, useMemo } from 'react';
```
Comment (guanbinrui) [inline src/app/(normal)/profile/[source]/[id]/page.tsx:6]: Didn't used.

File: src/components/Compose/Media.tsx [hunk 2]
```diff
@@ -95,7 +95,7 @@ export default function Media({ close }: MediaProps) {
                 <div
                     className={classNames(
                         'flex h-8 items-center gap-2',
-                        video ? 'cursor-not-allowed' : 'cursor-pointer hover:bg-bg',
+                        video || currentFarcasterProfile ? 'cursor-not-allowed opacity-50' : 'cursor-pointer hover:bg-bg',
```
Comment (guanbinrui) [inline src/components/Compose/Media.tsx:98]: Do you mean this?

```suggestion
                        video && currentFarcasterProfile ? 'cursor-not-allowed opacity-50' : 'cursor-pointer hover:bg-bg',
```

File: src/components/Posts/ActionType.tsx [hunk 3]
```diff
@@ -97,7 +101,7 @@ export const FeedActionType = memo<FeedActionType>(function FeedActionType({ pos
                 <div className="mb-3 flex items-center space-x-2 text-[15px] text-secondary">
                     <MirrorIcon width={16} height={16} />
                     <Link href={getProfileUrl(first(post.mirrors)!)}>
-                        <strong>{first(post.mirrors)?.displayName}</strong>{' '}
+                        <strong>{first(post.mirrors)?.profileId === currentProfile?.profileId ? t`You` : first(post.mirrors)?.displayName}</strong>{' '}
```
Comment (guanbinrui) [inline src/components/Posts/ActionType.tsx:104]: Use `isSameProfile` for profiles comparison.

File: src/components/Posts/ActionType.tsx [hunk 4]
```diff
@@ -107,7 +111,7 @@ export const FeedActionType = memo<FeedActionType>(function FeedActionType({ pos
                     {post.hasLiked ? <LikedIcon width={17} height={16} /> : <LikeIcon width={17} height={16} />}
                     <Link href={getProfileUrl(first(post.reactions)!)}>
                         <Trans>
-                            <strong>{first(post.reactions)?.displayName}</strong> liked
+                            <strong>{first(post.mirrors)?.profileId === currentProfile?.profileId ? t`You` : first(post.mirrors)?.displayName}</strong> liked
```
Comment (guanbinrui) [inline src/components/Posts/ActionType.tsx:114]: Use `isSameProfile` for profiles comparison.

File: src/hooks/useIsLogin.ts [hunk 5]
```diff
@@ -9,18 +9,19 @@ export function useIsLogin(source?: SocialPlatform) {
     const account = useAccount();
     const currentLensProfile = useLensStateStore.use.currentProfile();
     const currentFarcasterProfile = useFarcasterStateStore.use.currentProfile();
+    const isLensLogin = currentLensProfile?.signless ? currentLensProfile.profileId : (currentLensProfile?.profileId && currentLensProfile.ownedBy?.address === account.address)
```
Comment (guanbinrui) [inline src/hooks/useIsLogin.ts:12]: Use `isSam Address` for addresses comparison.

File: src/providers/types/LensGraphql/generated.ts [hunk 6]
```diff
@@ -0,0 +1,758 @@
+import { gql } from '@apollo/client'
```
Comment (guanbinrui) [inline src/providers/types/LensGraphql/generated.ts:1]: This file shouldn't be tracked. Include a generation process in CI.

File: src/components/Compose/Media.tsx (hunk 2 above)
Comment (beyond009) [reply src/components/Compose/Media.tsx:98]: This button need to be disabled when video or currentFarcasterProfile is not null

File: src/providers/types/LensGraphql/generated.ts (hunk 6 above)
Comment (beyond009) [reply src/providers/types/LensGraphql/generated.ts:1]: This file is copied from lens protocol repo

File: src/modals/LogoutModal.tsx [hunk 7]
```diff
@@ -112,6 +115,7 @@ export const LogoutModal = forwardRef<SingletonModalRefCreator<LogoutModalProps
                                                     clearFarcasterProfile();
                                                     break;
                                             }
+                                            router.push('/');
                                             dispatch?.close();
```
Comment (guanbinrui) [inline src/modals/LogoutModal.tsx:119]:
```suggestion
                                            dispatch?.close();
                                            await delay(300);
                                            router.push('/');
```

File: src/providers/lens/SocialMedia.ts [hunk 8]
```diff
@@ -286,8 +291,20 @@ export class LensSocialMedia implements Provider {
...
+        const request: ProfilesManagedRequest | LastLoggedInProfileRequest = {
+            for: address,
+        };
+        const {
+            data: { lastLoggedInProfile },
+        } = await profilesManagedQuery(request);
         const profiles = await this.client.wallet.profilesManaged({ for: address });
-        return profiles.items.map(formatLensProfile);
+        const result = profiles.items.map(formatLensProfile);
+        const index = result.findIndex((profile) => profile.handle === lastLoggedInProfile?.handle?.fullHandle);
```
Comment (guanbinrui) [inline src/providers/lens/SocialMedia.ts:302]: `isSameProfile`

File: src/providers/lens/SocialMedia.ts [hunk 9]
```diff
@@ -286,8 +291,20 @@ export class LensSocialMedia implements Provider {
...
+        if (index > -1) {
+            const [value] = result.splice(index, 1);
+            result.unshift(value);
```
Comment (guanbinrui) [inline src/providers/lens/SocialMedia.ts:305]: You can use `Array.prototype.sort`.

File: src/components/Profile/ContentTabs.tsx [hunk 10]
```diff
@@ -43,9 +44,19 @@ export default function ContentTabs({ profileId, source }: ContentTabsProps) {
                 ))}
             </div>
 
-            {tab === TabEnum.Feed && <ContentFeed source={source} profileId={profileId} />}
+            {tab === TabEnum.Feed && (
+                <Suspense fallback={<Loading />}>
+                    {' '}
+                    <ContentFeed source={source} profileId={profileId} />{' '}
+                </Suspense>
```
Comment (guanbinrui) [inline src/components/Profile/ContentTabs.tsx:51]:
```suggestion
                <Suspense fallback={<Loading />}>
                    <ContentFeed source={source} profileId={profileId} />{' '}
                </Suspense>
```

File: src/components/Profile/ContentTabs.tsx [hunk 11]
```diff
@@ -43,9 +44,19 @@ export default function ContentTabs({ profileId, source }: ContentTabsProps) {
...
+            )}
 
-            {tab === TabEnum.Collected && <ContentCollected source={source} profileId={profileId} />}
+            {tab === TabEnum.Collected && (
+                <Suspense fallback={<Loading />}>
+                    {' '}
+                    <ContentCollected source={source} profileId={profileId} />{' '}
+                </Suspense>
```
Comment (guanbinrui) [inline src/components/Profile/ContentTabs.tsx:58]:
```suggestion
                <Suspense fallback={<Loading />}>
                    <ContentCollected source={source} profileId={profileId} />{' '}
                </Suspense>
```

File: src/components/Posts/ActionType.tsx [hunk 12]
```diff
@@ -107,7 +116,12 @@ export const FeedActionType = memo<FeedActionType>(function FeedActionType({ pos
                     {post.hasLiked ? <LikedIcon width={17} height={16} /> : <LikeIcon width={17} height={16} />}
                     <Link href={getProfileUrl(first(post.reactions)!)}>
                         <Trans>
-                            <strong>{first(post.reactions)?.displayName}</strong> liked
+                            <strong>
+                                {isSameProfile(first(post.mirrors), currentProfile)
+                                    ? t`You`
+                                    : first(post.mirrors)?.displayName}
+                            </strong>{' '}
+                            liked
```
Comment (guanbinrui) [inline src/components/Posts/ActionType.tsx:124]: Remove `<Trans />`.

```suggestion
                            <strong>
                                {isSameProfile(first(post.mirrors), currentProfile)
                                    ? t`You`
                                    : first(post.mirrors)?.displayName}
                            </strong>{' '}
                           t`liked`
```

File: src/components/Posts/ActionType.tsx [hunk 13]
```diff
@@ -97,7 +102,11 @@ export const FeedActionType = memo<FeedActionType>(function FeedActionType({ pos
                 <div className="mb-3 flex items-center space-x-2 text-[15px] text-secondary">
                     <MirrorIcon width={16} height={16} />
                     <Link href={getProfileUrl(first(post.mirrors)!)}>
-                        <strong>{first(post.mirrors)?.displayName}</strong>{' '}
+                        <strong>
+                            {isSameProfile(first(post.mirrors), currentProfile)
+                                ? t`You`
+                                : first(post.mirrors)?.displayName}
+                        </strong>{' '}
```
Comment (guanbinrui) [inline src/components/Posts/ActionType.tsx:109]: Use `<Trans />`.

```suggestion
                        <strong>
                            {isSameProfile(first(post.mirrors), currentProfile)
                                ? t`You`
                                : first(post.mirrors)?.displayName}
                        </strong>{' '}
```

File: src/components/Posts/ActionType.tsx [hunk 14]
```diff
@@ -97,7 +102,11 @@ export const FeedActionType = memo<FeedActionType>(function FeedActionType({ pos
                 <div className="mb-3 flex items-center space-x-2 text-[15px] text-secondary">
                     <MirrorIcon width={16} height={16} />
                     <Link href={getProfileUrl(first(post.mirrors)!)}>
-                        <strong>{first(post.mirrors)?.displayName}</strong>{' '}
+                        <strong>
+                            {isSameProfile(first(post.mirrors), currentProfile)
```
Comment (guanbinrui) [inline src/components/Posts/ActionType.tsx:106]: Avoid splitting `You ...` into multiple translation items.

File: src/components/Posts/ActionType.tsx [hunk 15]
```diff
@@ -107,7 +116,12 @@ export const FeedActionType = memo<FeedActionType>(function FeedActionType({ pos
                     {post.hasLiked ? <LikedIcon width={17} height={16} /> : <LikeIcon width={17} height={16} />}
                     <Link href={getProfileUrl(first(post.reactions)!)}>
                         <Trans>
-                            <strong>{first(post.reactions)?.displayName}</strong> liked
+                            <strong>
+                                {isSameProfile(first(post.mirrors), currentProfile)
+                                    ? t`You`
+                                    : first(post.mirrors)?.displayName}
+                            </strong>{' '}
```
Comment (guanbinrui) [inline src/components/Posts/ActionType.tsx:123]: Same here.

File: src/modals/LoginModal.tsx [hunk 16]
```diff
@@ -65,9 +74,18 @@ export const LoginModal = forwardRef<SingletonModalRefCreator<LoginModalProps |
...
                 }
             } catch (error) {
-                enqueueSnackbar(error instanceof Error ? error.message : t`Failed to connect wallet.`, {
-                    variant: 'error',
-                });
+                enqueueSnackbar(
+                    <div>
+                        <span className="font-bold">
+                            <Trans>Connection failed</Trans>
+                        </span>{' '}
```
Comment (guanbinrui) [inline src/modals/LoginModal.tsx:81]:
```suggestion
                        </span>
```

File: src/modals/LoginModal.tsx [hunk 17]
```diff
@@ -47,9 +47,18 @@ export const LoginModal = forwardRef<SingletonModalRefCreator<LoginModalProps |
...
                         });
                         if (!profiles.length) {
-                            enqueueSnackbar(t`No Lens profile found. Please change to another wallet.`, {
-                                variant: 'error',
-                            });
+                            enqueueSnackbar(
+                                <div>
+                                    <span className="font-bold">
+                                        <Trans>Wrong wallet</Trans>
+                                    </span>{' '}
```
Comment (guanbinrui) [inline src/modals/LoginModal.tsx:54]:
```suggestion
                                    </span>
```

File: src/components/Posts/ActionType.tsx [hunk 18]
```diff
@@ -88,7 +93,11 @@ export const FeedActionType = memo<FeedActionType>(function FeedActionType({ pos
                     <MirrorIcon width={16} height={16} />
                     <Link href={getProfileUrl(post.reporter)}>
                         <Trans>
-                            <strong>{post.reporter.displayName}</strong> mirrored
+                            <strong>
+                                {' '}
```
Comment (guanbinrui) [inline src/components/Posts/ActionType.tsx:97]:
```suggestion
```

File: src/components/Posts/ActionType.tsx [hunk 19]
```diff
@@ -97,8 +105,14 @@ export const FeedActionType = memo<FeedActionType>(function FeedActionType({ pos
...
                     <MirrorIcon width={16} height={16} />
                     <Link href={getProfileUrl(first(post.mirrors)!)}>
-                        <strong>{first(post.mirrors)?.displayName}</strong>{' '}
-                        {post.source === SocialPlatform.Farcaster ? t`recasted` : t`mirrored`}
+                        <Trans>
+                            <strong>
+                                {isSameProfile(first(post.mirrors), currentProfile)
+                                    ? 'You'
+                                    : first(post.mirrors)?.displayName}
+                            </strong>
```
Comment (guanbinrui) [inline src/components/Posts/ActionType.tsx:113]:
```suggestion
                            </strong>{' '}
```

File: src/components/Posts/ActionType.tsx [hunk 20]
```diff
@@ -107,7 +121,12 @@ export const FeedActionType = memo<FeedActionType>(function FeedActionType({ pos
                     {post.hasLiked ? <LikedIcon width={17} height={16} /> : <LikeIcon width={17} height={16} />}
                     <Link href={getProfileUrl(first(post.reactions)!)}>
                         <Trans>
-                            <strong>{first(post.reactions)?.displayName}</strong> liked
+                            <strong>
+                                {isSameProfile(first(post.mirrors), currentProfile)
+                                    ? `You`
+                                    : first(post.mirrors)?.displayName}
+                            </strong>
```
Comment (guanbinrui) [inline src/components/Posts/ActionType.tsx:128]:
```suggestion
                            </strong>{' '}
```

File: src/components/Profile/ContentTabs.tsx [hunk 21]
```diff
@@ -43,9 +44,17 @@ export default function ContentTabs({ profileId, source }: ContentTabsProps) {
                 ))}
             </div>
 
-            {tab === TabEnum.Feed && <ContentFeed source={source} profileId={profileId} />}
+            {tab === TabEnum.Feed && (
+                <Suspense fallback={<Loading />}>
+                    <ContentFeed source={source} profileId={profileId} />{' '}
```
Comment (guanbinrui) [inline src/components/Profile/ContentTabs.tsx:49]:
```suggestion
                    <ContentFeed source={source} profileId={profileId} />
```

File: src/components/Profile/ContentTabs.tsx [hunk 22]
```diff
@@ -43,9 +44,17 @@ export default function ContentTabs({ profileId, source }: ContentTabsProps) {
...
+                </Suspense>
+            )}
 
-            {tab === TabEnum.Collected && <ContentCollected source={source} profileId={profileId} />}
+            {tab === TabEnum.Collected && (
+                <Suspense fallback={<Loading />}>
+                    <ContentCollected source={source} profileId={profileId} />{' '}
```
Comment (guanbinrui) [inline src/components/Profile/ContentTabs.tsx:55]:
```suggestion
                    <ContentCollected source={source} profileId={profileId} />
```

File: src/components/Posts/ActionType.tsx [hunk 23]
```diff
@@ -107,7 +121,12 @@ export const FeedActionType = memo<FeedActionType>(function FeedActionType({ pos
                     {post.hasLiked ? <LikedIcon width={17} height={16} /> : <LikeIcon width={17} height={16} />}
                     <Link href={getProfileUrl(first(post.reactions)!)}>
                         <Trans>
-                            <strong>{first(post.reactions)?.displayName}</strong> liked
+                            <strong>
+                                {isSameProfile(first(post.mirrors), currentProfile)
+                                    ? `You`
```
Comment (guanbinrui) [inline src/components/Posts/ActionType.tsx:126]:
```suggestion
                                    ? t`You`
```

File: src/components/Posts/ActionType.tsx [hunk 24]
```diff
@@ -97,8 +105,14 @@ export const FeedActionType = memo<FeedActionType>(function FeedActionType({ pos
                 <div className="mb-3 flex items-center space-x-2 text-[15px] text-secondary">
                     <MirrorIcon width={16} height={16} />
                     <Link href={getProfileUrl(first(post.mirrors)!)}>
-                        <strong>{first(post.mirrors)?.displayName}</strong>{' '}
-                        {post.source === SocialPlatform.Farcaster ? t`recasted` : t`mirrored`}
+                        <Trans>
+                            <strong>
+                                {isSameProfile(first(post.mirrors), currentProfile)
+                                    ? 'You'
```
Comment (guanbinrui) [inline src/components/Posts/ActionType.tsx:111]:
```suggestion
                                    ? t`You`
```

File: src/components/Posts/ActionType.tsx [hunk 25]
```diff
@@ -88,7 +93,10 @@ export const FeedActionType = memo<FeedActionType>(function FeedActionType({ pos
                     <MirrorIcon width={16} height={16} />
                     <Link href={getProfileUrl(post.reporter)}>
                         <Trans>
-                            <strong>{post.reporter.displayName}</strong> mirrored
+                            <strong>
+                                {isSameProfile(post.reporter, currentProfile) ? t`You` : post.reporter.displayName}
+                            </strong>
```
Comment (guanbinrui) [inline src/components/Posts/ActionType.tsx:98]:
```suggestion
                            </strong>{' '}
```

File: src/modals/LoginModal.tsx [hunk 26]
```diff
@@ -65,9 +74,18 @@ export const LoginModal = forwardRef<SingletonModalRefCreator<LoginModalProps |
...
-                enqueueSnackbar(error instanceof Error ? error.message : t`Failed to connect wallet.`, {
-                    variant: 'error',
-                });
+                enqueueSnackbar(
+                    <div>
+                        <span className="font-bold">
+                            <Trans>Connection failed</Trans>
+                        </span>
+                        <br />
+                        <Trans>Oops… Something went wrong. Please try again</Trans>
```
Comment (guanbinrui) [inline src/modals/LoginModal.tsx:83]:
```suggestion
                        <Trans>Oops… Something went wrong. Please try again.</Trans>
```

File: src/modals/LoginModal.tsx [hunk 27]
```diff
@@ -47,9 +47,18 @@ export const LoginModal = forwardRef<SingletonModalRefCreator<LoginModalProps |
...
-                            enqueueSnackbar(t`No Lens profile found. Please change to another wallet.`, {
-                                variant: 'error',
-                            });
+                            enqueueSnackbar(
+                                <div>
+                                    <span className="font-bold">
+                                        <Trans>Wrong wallet</Trans>
+                                    </span>
+                                    <br />
+                                    <Trans>No Lens profile found. Please change another wallet</Trans>
```
Comment (guanbinrui) [inline src/modals/LoginModal.tsx:56]:
```suggestion
                                    <Trans>No Lens profile found. Please change another wallet.</Trans>
```

File: src/locales/en/messages.po [hunk 28]
```diff
@@ -599,7 +614,10 @@ msgid "Following"
 msgstr "Following"
 
 msgid "Dark mode"
-msgstr "Dark mode"
+msgstr "Dark mode<<<<<<< HEAD======="
```
Comment (guanbinrui) [inline src/locales/en/messages.po:617]: ！

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**